python/src/equistore/operations/lstsq.py

```diff
@@ -9,17 +9,85 @@
 
 
 def lstsq(X: TensorMap, Y: TensorMap, rcond, driver=None) -> TensorMap:
-    """
+    r"""
     Solve a linear system using two :py:class:`TensorMap`.
 
-    Return the least-squares solution to a linear equation ``Y = X * w`` solving
-    for ``w``, where ``Y``, ``X`` and ``w`` are all :py:class:`TensorMap`. ``Y``
-    and ``X`` must have the same ``keys``.
+    The least-squares solution ``w_b`` for the linear system
+    :math:`X_b w_b = Y_b` is solved for all blocks :math:`b` in ``X``
+    and ``Y``. ``X`` and ``Y`` must have the same keys.
+    The returned :py:class:`TensorMap` ``w`` has the same keys as ``X`` and
+    ``Y``, and stores in each block the least-squares solutions :math:`w_b`.
+
+    If a block has multiple components, they are moved to the "samples" axis
+    before solving the linear system.
+
+    If gradients are present, they must be present in both ``X`` and ``Y``.
+    Gradients are concatenated with the block values along the "samples" axis,
+    :math:`A_b = [X_b, {\nabla} X_b]`, :math:`B_b = [Y_b, {\nabla} Y_b]`, and the
+    linear system :math:`A_b w_b = B_b` is solved for :math:`w_b` using
+    least-squares.
+
+    .. note::
+      The solutions :math:`w_b` differ from the output of numpy or torch in that they
+      are already transposed. Be aware of that if you want to manually access
+      the values of blocks of ``w`` (see also the example below).
+
+    Here is an example using this function:
+
+    >>> import numpy as np
+    >>> from equistore import Labels, TensorBlock, TensorMap
+    >>> from equistore import lstsq
+    ...
+    >>> values_X = np.array([
+    ...     [1.0, 2.0],
+    ...     [3.0, 1.0],
+    ... ])
+    ...
+    >>> values_Y = np.array([
+    ...     [1.0, 0.0],
+    ...     [0.0, 1.0],
+    ... ])
+    ...
+    >>> samples = Labels(
+    ...     ["structure"],
+    ...     np.array([[0], [1]]),
+    ... )
+    ...
+    >>> components = []
+    ...
+    >>> properties = Labels(
+    ...     ["properties"],
+    ...     np.array([[0], [1]])
+    ... )
+    ...
+    >>> keys = Labels(names=["key"], values=np.array([[0]]))
+    ...
+    >>> block_X = TensorBlock(
+    ...     values_X, samples, components, properties,
+    ... )
+    ...
+    >>> block_Y = TensorBlock(
+    ...     values_Y, samples, components, properties,
+    ... )
+    ...
+    >>> X = TensorMap(keys, [block_X])
+    ...
+    >>> Y = TensorMap(keys, [block_Y])
+    ...
+    >>> w = lstsq(X, Y, rcond=1e-10)
+    ...
+    >>> # Note: we take the transpose here
+    >>> print(X.block(0).values @ w.block(0).values.T)
+    [[ 1.00000000e+00 -2.22044605e-16]
+     [-3.33066907e-16  1.00000000e+00]]
```

Comment:
This is failing on CI since the 1e-16 values are just noise. Maybe before printing you could replace values below 1e-15 with 0? something like 

```
y = X.block(0).values @ w.block(0).values.T
y[y.abs() < 1e-15] = 0.0
```

Reply:
Sure!